scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,123 @@
+"""
+This script is referenced by TerraForm as an external data source.
+Prints the current deployment's Elasticsearch node IDs as JSON to STDOUT. If
+the list of node IDs is less than the number of placeholders on the Cloudwatch
+dashboard template, the output is padded with 'blank' entries.
+Example:
+{"nodes": "tVsGH2BSXXYe9sMLEYRcIv,ydG9meZMAGkTnAlPgLljz8,blank,blank"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+import re
+import sys
+from time import (
+    sleep,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to retry the request to AWS if an error is encountered.
+max_attempts = 10
+
+
+def main() -> int:
+    i = iter(range(max_attempts))
+    while True:
+        attempt = next(i, None)
+        if attempt is None:
+            log.debug('Error: Max attempts exhausted')
+            return 1
+        else:
+            try:
+                response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                       MetricName='CPUUtilization',
+                                                       RecentlyActive='PT3H',
+                                                       Dimensions=[
+                                                           {
+                                                               'Name': 'DomainName',
+                                                               'Value': config.es_domain
+                                                           }
+                                                       ])
+            except ClientError as e:
+                log.debug('Error: ' + e.response['Error']['Code'] + '. Retrying...')
+                sleep(attempt + 1)
```

Comment:
A linear back-off is unusual and has odd behavior when things get stuck. They cause too many requests early in a stuck state, and too few request later. Exponential back-off with a cap is what is typically used. Here you can use fixed 5s back-off.